apps/dav/lib/Upload/UploadHome.php

```diff
@@ -97,10 +98,43 @@ private function getView() {
 		$rootView = new View();
 		$user = \OC::$server->getUserSession()->getUser();
 		Filesystem::initMountPoints($user->getUID());
-		if (!$rootView->file_exists('/' . $user->getUID() . '/uploads')) {
-			$rootView->mkdir('/' . $user->getUID() . '/uploads');
+
+		$config = \OC::$server->get(IConfig::class);
+
+		$allowSymlinks = $config->getSystemValueBool('localstorage.allowsymlinks', false);
+		$uploadsPath = $config->getSystemValueString('uploads_path', '');
+		$userPath = '/' . $user->getUID() . '/uploads';
+        $absoluteUserPath = $rootView->getLocalFile($userPath);
+
+		if ($allowSymlinks && $uploadsPath !== '' && $absoluteUserPath) {
+			$uploadUserPath = $uploadsPath . $userPath;
+
+			if (!$rootView->file_exists($userPath) || !is_link($absoluteUserPath) || ($uploadUserPath != realpath($absoluteUserPath)) ) {
+
+                if (!is_dir($uploadUserPath)) {
+                    mkdir($uploadUserPath, 0750, true);
+                }
+
+				// useful if link is broken due to $uploadUserPath changes
+                if (is_link($absoluteUserPath)) {
+                    unlink($absoluteUserPath);
+                } elseif (is_dir($absoluteUserPath)) {
+                    $rootView->rmdir($userPath);
+                }
+                symlink($uploadUserPath, $absoluteUserPath);
+
+			}
+
+		} else {
+            if ($absoluteUserPath && is_link($absoluteUserPath)) {
```

Comment:
More explicit check?
```suggestion
            if ($absoluteUserPath !== null && is_link($absoluteUserPath)) {
```